bevy_lint/src/lints/restriction/missing_trait_impls.rs

```diff
@@ -0,0 +1,144 @@
+use clippy_utils::{diagnostics::span_lint_hir_and_then, sugg::DiagExt, ty::implements_trait};
+use rustc_errors::Applicability;
+use rustc_hir::{ItemKind, def_id::DefId};
+use rustc_lint::{LateContext, LateLintPass, Lint};
+
+use crate::{declare_bevy_lint, declare_bevy_lint_pass, utils::traits::TraitType};
+
+declare_bevy_lint! {
+    pub(crate) MISSING_DEFAULT,
+    super::Restriction,
+    "defined a unit component without a `Default` implementation",
+}
+
+declare_bevy_lint! {
+    pub(crate) MISSING_CLONE,
+    super::Restriction,
+    "defined a unit component without a `Clone` implementation",
+}
+
+declare_bevy_lint! {
+    pub(crate) MISSING_COPY,
+    super::Restriction,
+    "defined a unit component without a `Copy` implementation",
+}
+
+declare_bevy_lint_pass! {
+    pub(crate) MissingTraitImpls => [MISSING_DEFAULT,MISSING_CLONE,MISSING_COPY],
+}
+
+impl<'tcx> LateLintPass<'tcx> for MissingTraitImpls {
+    fn check_crate(&mut self, cx: &LateContext<'tcx>) {
+        // Finds all types that implement `Component` in this crate.
+        let components: Vec<TraitType> =
+            TraitType::from_local_crate(cx, &crate::paths::COMPONENT).collect();
+
+        for component in components {
+            // Skip if a types originates from a foreign crate's macro
+            if component
+                .item_span
+                .in_external_macro(cx.tcx.sess.source_map())
+            {
+                continue;
+            }
+
+            let def_id = component.hir_id.owner.to_def_id();
+
+            // Skip binder as we are not interested in the generics
```

Comment:
```suggestion
            // Skip binder as unit types cannot have generics.
```

(Nit) This makes it more clear why we don't need to handle generics or lifetimes :)